### PR TITLE
Allow to enable frame locking in the markers/markers arrays

### PR DESCRIPTION
**Pull request by VictorLamoine**
This allows the markers to be automatically updated as the base frame moves without having to republish.

In short: this makes RVT dynamic (if enabled)

### CODE REVIEW

File: src/rviz_visual_tools.cpp [hunk 1]
```diff
@@ -896,6 +902,10 @@ bool RvizVisualTools::publishMarkers(visualization_msgs::MarkerArray &markers)
   if (markers.markers.empty())
     return false;
 
+  if (frame_locking_enabled_)
+    for (auto &marker: markers.markers)
+      marker.frame_locked = true;
```
Comment (davetcoleman) [inline src/rviz_visual_tools.cpp]: Why is the for loop necessary when all markers should already have this property enabled above where you have:

> marker.frame_locked = frame_locking_enabled_;

?

Comment (VictorLamoine) [inline src/rviz_visual_tools.cpp]: I thought there were two separate markers type but I was wrong.
This was redundant.

I removed the loop and updated my PR.

**Comment by VictorLamoine** [comment]
Here is a GIF:

peek 03-08-2017 10-39

**Comment by davetcoleman** [comment]
Can you add your GIF to the README.md and a sentence saying what feature enables that? (frame locking)

**Comment by VictorLamoine** [comment]
Done;
https://github.com/VictorLamoine/rviz_visual_tools#helper-functions

**Comment by felixvd** [comment]
Damn. This should be in the top-level Visualization Marker documentation.

**Pull request merged**